types/block.go

```diff
@@ -1363,11 +1375,39 @@ func DataFromProto(dp *cmtproto.Data) (Data, error) {
 		data.Txs = Txs{}
 	}
 
+	data.DataRootHash = dp.Hash
+	data.SquareSize = dp.SquareSize
```

Comment:
[blocking]

since we're not setting cached hash here, will we break any calls to `data.Hash()`? 

like here
https://github.com/celestiaorg/celestia-core/blob/a2b8ce5ddfe579baaf33230843d7b234e09bd15d/light/rpc/client.go#L736-L739

Reply:
i reverted to how it was before, the flow here seems like it would produce different to how it was before. I.e. override the value we set.